rules-tattletale/addon/src/main/java/org/jboss/windup/rules/apps/tattletale/TattletaleRuleProvider.java

```diff
@@ -58,8 +61,13 @@ public Configuration getConfiguration(RuleLoaderContext ruleLoaderContext)
         @Override
         public void perform(GraphRewrite event, EvaluationContext context)
         {
-            Boolean generateReport = (Boolean) event.getGraphContext().getOptionMap().get(EnableTattletaleReportOption.NAME);
-            if (generateReport == null || !generateReport)
+            Boolean enableReport = (Boolean) event.getGraphContext().getOptionMap().get(EnableTattletaleReportOption.NAME);
+            Boolean disableReport = (Boolean) event.getGraphContext().getOptionMap().get(DisableTattletaleReportOption.NAME);
+            Collection<String> targets = (Collection<String>) event.getGraphContext().getOptionMap().get(TargetOption.NAME);
+            boolean eapTarget = targets != null && targets.stream().anyMatch(target -> target.startsWith("eap"));
+            if (eapTarget && BooleanUtils.isTrue(disableReport) && BooleanUtils.isNotTrue(enableReport))
```

Comment:
As this is not a straight forward condition, what do you think on extracting a method in charge of decide if the report is enabled ?